src/js/legend.js

```diff
@@ -25,19 +25,61 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
             // dataSet accepts:
             // 1. an array of objects. Must contain "id", "value" and "label" variables.
             // Example: [{id: string, value: number, label: string} ... ]
-            dataSet: []
+            dataSet: [],
+            total: {
+                // value: number,
+                // percentage: number
+            }
         },
+        modelRelay: [{
```

Comment:
@waharnum, I agree having the model relay in each component helps to maintain their independencies. I don't have a good idea of maintaining the independency while reducing the repetition. 

@amb26, any thoughts?